### PR TITLE
feat(NetworkFirewall): New Service and Check

### PR DESCRIPTION
**Pull request by gabrielsoltz**
### Context

Ahora si, one more! New `networkfirewall` service!
Check: networkfirewall_in_all_vpc (medium): Ensure all VPCs have Network Firewall enabled

Without a network firewall, it can be difficult to monitor and control traffic within the VPC. This can make it harder to detect and prevent attacks or unauthorized access to resources.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/networkfirewall/networkfirewall_service.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                        self.network_firewalls.append(
+                            Firewall(
+                                arn=network_firewall["FirewallArn"],
+                                region=regional_client.region,
+                                name=network_firewall["FirewallName"],
+                            )
+                        )
+        except Exception as error:
+            logger.error(
+                f"{error.__class__.__name__}:{error.__traceback__.tb_lineno} -- {error}"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/networkfirewall/networkfirewall_service.py]: Can you add the region of the regional_client in the error?

File: prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.metadata.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "Severity": "medium",
+  "ResourceType": "Other",
+  "Description": "Ensure all VPCs have Network Firewall enabled",
+  "Risk": "Without a network firewall, it can be difficult to monitor and control traffic within the VPC. This can make it harder to detect and prevent attacks or unauthorized access to resources.",
+  "RelatedUrl": "https://docs.aws.amazon.com/network-firewall/latest/developerguide/setting-up.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "aws network-firewall list-firewalls",
+      "NativeIaC": "",
+      "Other": "",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.metadata.json]:
```suggestion
      "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/NetworkFirewall/network-firewall-in-use.html",
```

File: prowler/providers/aws/services/networkfirewall/networkfirewall_service.py (hunk 1 above)
Comment (gabrielsoltz) [reply prowler/providers/aws/services/networkfirewall/networkfirewall_service.py]: changed ✅

File: prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.metadata.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "aws",
+  "CheckID": "networkfirewall_in_all_vpc",
+  "CheckTitle": "Ensure all VPCs have Network Firewall enabled",
+  "CheckType": [],
+  "ServiceName": "network-firewall",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:partition:service::account-id:organization/organization-id",
```
Comment (jfagoagas) [inline prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.metadata.json]: Change this please!

File: prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.metadata.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  "Severity": "medium",
+  "ResourceType": "Other",
+  "Description": "Ensure all VPCs have Network Firewall enabled",
+  "Risk": "Without a network firewall, it can be difficult to monitor and control traffic within the VPC. This can make it harder to detect and prevent attacks or unauthorized access to resources.",
+  "RelatedUrl": "https://docs.aws.amazon.com/network-firewall/latest/developerguide/setting-up.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "aws network-firewall list-firewalls",
```
Comment (jfagoagas) [inline prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.metadata.json]: Do you have the command to enable it?

File: prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+
+class networkfirewall_in_all_vpc(Check):
+    def execute(self):
+        findings = []
+        for vpc in vpc_client.vpcs:
+            report = Check_Report_AWS(self.metadata())
+            report.region = vpc.region
+            report.resource_id = vpc.id
+            report.resource_arn = ""
```
Comment (jfagoagas) [inline prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.py:15]: Could you please fill this ARN?

File: prowler/providers/aws/services/networkfirewall/networkfirewall_service.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                    if not self.audit_resources or (
+                        is_resource_filtered(
+                            network_firewall["FirewallArn"], self.audit_resources
+                        )
+                    ):
+                        self.network_firewalls.append(
+                            Firewall(
+                                arn=network_firewall["FirewallArn"],
+                                region=regional_client.region,
+                                name=network_firewall["FirewallName"],
```
Comment (jfagoagas) [inline prowler/providers/aws/services/networkfirewall/networkfirewall_service.py]: Please use the `dict.get()` syntax here.

File: prowler/providers/aws/services/networkfirewall/networkfirewall_service.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+                regional_client = self.regional_clients[network_firewall.region]
+                describe_firewall = regional_client.describe_firewall(
+                    FirewallArn=network_firewall.arn
+                )["Firewall"]
+                network_firewall.policy_arn = describe_firewall["FirewallPolicyArn"]
+                network_firewall.vpc_id = describe_firewall["VpcId"]
+                network_firewall.tags = describe_firewall["Tags"]
+                network_firewall.encryption_type = describe_firewall[
+                    "EncryptionConfiguration"
+                ]["Type"]
```
Comment (jfagoagas) [inline prowler/providers/aws/services/networkfirewall/networkfirewall_service.py]: Please use the `dict.get()` syntax here.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.py (hunk 5 above)
Comment (gabrielsoltz) [reply prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.py:15]: with what? VPC object doesn't provide arn.

File: prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.metadata.json (hunk 4 above)
Comment (gabrielsoltz) [reply prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.metadata.json]: added it ✅

File: prowler/providers/aws/services/networkfirewall/networkfirewall_service.py (hunk 6 above)
Comment (gabrielsoltz) [reply prowler/providers/aws/services/networkfirewall/networkfirewall_service.py]: changed ✅

File: prowler/providers/aws/services/networkfirewall/networkfirewall_service.py (hunk 7 above)
Comment (gabrielsoltz) [reply prowler/providers/aws/services/networkfirewall/networkfirewall_service.py]: changed ✅

File: prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.py (hunk 5 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/networkfirewall/networkfirewall_in_all_vpc/networkfirewall_in_all_vpc.py:15]: Good point!

**Review by jfagoagas — APPROVED** [review]
Good to go! 🚀

**Pull request merged**